docs/howto/features/imagebuilding.md

```diff
@@ -0,0 +1,153 @@
+
+```yaml
+singleuser:
+  profileList:
+    - display_name: "Only Profile Available, this info is not shown in the UI"
+      slug: only-choice
+      profile_options:
+        image:
+          display_name: Image
+          unlisted_choice:
+            enabled: True
+            display_name: "Custom image"
+            validation_regex: "^.+:.+$"
+            validation_message: "Must be a publicly available docker image, of form <image-name>:<tag>"
+            display_name_in_choices: "Specify an existing docker image"
+            description_in_choices: "Use a pre-existing docker image from a public docker registry (dockerhub, quay, etc)"
+            kubespawner_override:
+              image: "{value}"
+          choices:
+            pangeo:
+              display_name: Pangeo Notebook Image
+              description: "Python image with scientific, dask and geospatial tools"
+              kubespawner_override:
+                image: pangeo/pangeo-notebook:2023.09.11
+            scipy:
+              display_name: Jupyter SciPy Notebook
+              slug: scipy
+              kubespawner_override:
+                image: jupyter/scipy-notebook:2023-06-26
+```
+
+## The `binderhub-service` chart
+
+We will use the [binderhub-service](https://github.com/2i2c-org/binderhub-service/) Helm chart to run BinderHub, the Python software, as a standalone service to build and push images with [repo2docker](https://github.com/jupyterhub/repo2docker), next to JupyterHub.
+
+### How it works
+
+The `binderhub-service` installation starts a `docker-api` pod on each of the user nodes via the following [DaemonSet definition](https://github.com/2i2c-org/binderhub-service/blob/main/binderhub-service/templates/docker-api/daemonset.yaml).
+
+The `docker-api` pod setups and starts the [dockerd](https://docs.docker.com/engine/reference/commandline/dockerd/) daemon, that will then be accessible via a mounted unix socket, by the `build pods`.
+
+The `build pods` are created as a result of an image build request, and they must run on the same node as the builder pods to make use of the docker daemon. These pods mount a k8s Secret with the docker config file holding the necessary registry credentials.
+
+### How to set it up
+
+1. Setup a registry and safely store the credentials to push to it
+
+  - If the hub will be deployed on a GCP cluster, follow the steps 4 to 8 from the [binderhub-service installation guide](https://github.com/2i2c-org/binderhub-service?tab=readme-ov-file#installation). When you're done, you should have a gcr.io registry setup and the appropriate credentials to push images to it.
+
+  - If the hub will be deployed on another cloud provider than GCP, then you should setup the desired quay.io repository and robot account to push and pull from it. The steps below describes how to create one under the 2i2c quay.io organization, assuming this is where we wish the repositories that will be build by `binderhub-service` to be pushed.
```

Comment:
I would suggest that we actually create perhaps *one* new quay.io organization (like 2i2c-imagebuilding-hubs perhaps?) that can be used for this, instead of the existing 2i2c org. This strikes a decent balance between creating one new org per hub vs just re-using the same one we have.